server/src/controllers/christmas-tree/permits.es6

```diff
@@ -325,35 +284,35 @@ christmasTreePermits.generateRulesAndEmail = permit => permitSvgService.generate
  * @return {Object} - saved permit object
  */
 christmasTreePermits.getOnePermit = (req, res) => {
-  treesDb.christmasTreesPermits
-    .findOne({
-      where: {
-        permitId: req.params.id
-      },
-      include: [
-        {
-          model: treesDb.christmasTreesForests
-        }
-      ]
-    })
-    .then((permit) => {
-      if (permit && permit.status === 'Error') {
-        return getPermitError(res, permit);
-      } if (permit) {
-        const token = req.query.t;
-        jwt.verify(token, vcapConstants.PERMIT_SECRET, (err, decoded) => {
-          if (decoded) {
-            util.logControllerAction(req, 'christmasTreePermits.getOnePermit', permit);
-            return returnSavedPermit(res, permit);
+  const validToken = jwt.verify(req.query.t, vcapConstants.PERMIT_SECRET);
+  if (validToken) {
+    treesDb.christmasTreesPermits
+      .findOne({
+        where: {
+          permitId: req.params.id
+        },
+        include: [
+          {
+            model: treesDb.christmasTreesForests
           }
-          return res.status(404).send();
-        });
-      }
-      return res.status(404).send();
-    })
-    .catch((error) => {
-      util.handleErrorResponse(error, res, 'getOnePermit#end');
-    });
+        ]
+      })
+      .then((permit) => {
+        if (permit && permit.status === 'Error') {
+          return res.status(400).json(formatPermitError(permit));
+        }
+        if (permit) {
+          util.logControllerAction(req, 'christmasTreePermits.getOnePermit', permit);
```

Comment:
I've made a similar observation and agree in general. One thing to note is that the logged information includes the `permit` in this case, and other data in other cases, potentially complicating the implementation of the middlware. My thought is to delay action on this until more investigation can be done to fully understand the purpose of this logging.

Reply:
Yeah sounds good. If the "interface" for logControllerAction data is the request, then it might make sense to store data on the request. e.g. middleware that fetches a permit from the  DB, attaches it to the request, then calls `next()` for the controller and logControllerAction middleware to pull data directly from `req`.